util/flexcache.cc

```diff
@@ -45,7 +45,8 @@ FlexCache::FlexCache()
     // initialize total memory available based upon system data
     ret_val=getrlimit(RLIMIT_DATA, &limit);
 
-    if (0==ret_val && RLIM_INFINITY!=limit.rlim_max)
+    //  unsigned long caste to fix warning in smartos1.8, smartos 13.1, solaris10
+    if (0==ret_val && (unsigned long)RLIM_INFINITY!=limit.rlim_max)
```

Comment:
The problem with unconditionally casting RLIM_INFINITY to (unsigned long) is that the types of limit.rlim_max and (unsigned long) might be different sized integers. If one is 32-bit and the other 64-bit on some platform, then I'm not sure you'll get the comparison you want, since unsigned values typically get converted to signed integers during promotion (just depends on the compiler); then you might not get the comparison you intend due to sign extension.

Or this might "just work" on all the platforms we currently support.

If the only platforms complaining about this are the Solaris based compilers (as the comment indicates), could do the cast inside an "#ifdef SOLARIS" conditional block.